### ISSUE TITLE
Training AtomNas-a occur AssertionError

### ISSUE THREAD

**Issue by iFLYTEK-Embodied-Robotics-Team** [meijieru/AtomNAS#11]

**Comment by zhangyuan1994511** [comment]
@meijieru 
During check the training code, I found the constructed 'FLAGS._bn_to_prune' does not include the first 'mb.InvertedResidualChannels'(at train.py line 346), but during the shrink model process, the first 'mb.InvertedResidualChannels' has been included(at train.py line 48、53). It maybe the main reason to cause the above errors...

**Comment by zhangyuan1994511** [comment]
@meijieru
During check the training code, I found the constructed 'FLAGS._bn_to_prune' does not include the first 'mb.InvertedResidualChannels'(at train.py line 346), but during the shrink model process, the first 'mb.InvertedResidualChannels' has been included(at train.py line 48、53). It maybe the main reason to cause the above errors... Please check it~

**Comment by vtddggg** [comment]
Maybe my experience can help you. 
For me, I also run `atomnas_a` config without changing any code but choosing a smaller batch_size `per_gpu_batch_size: 32`
Now the training process runs at 74 epoch and no `AssertionError` happened.
![image](https://user-images.githubusercontent.com/27721248/82164497-8388cc00-98e3-11ea-9dee-6e4fc5851cc3.png)

So I think it probably caused by your batch_size.

**Comment by zhangyuan1994511** [comment]
Ok, I get it, Thank for your reply! I get it! But It is still seem a bug remian to be solved~

**Comment by zhangyuan1994511** [comment]
@vtddggg By the way, your training logging is not same as mine, yous training process seem not run the function: 'prune.output_searched_network', where occur the above 'AssertionError....'?

**Comment by vtddggg** [comment]
@zhangyuan1994511 can I see your complete training logging？

**Comment by zhangyuan1994511** [comment]
@vtddggg The full training log is too long, so I share it at Baidu cloud：https://pan.baidu.com/s/1ROwGvdTWBZUYfeb-MSwCRA ； saqk

**Comment by vtddggg** [comment]
ok, thank you! I checked it. But as my comparison with your log, our output looks the same😂
How you see I am not running the function: 'prune.output_searched_network'?

**Comment by zhangyuan1994511** [comment]
Maybe I make a mistake, The training code you have released outputs the pruned model, And any other reason about the  'AssertionError....'? If just the batch_size, it is still remian to solve~

**Comment by zhangyuan1994511** [comment]
@vtddggg 
Hi, Author! I'm sorry to bother you again. If a mobileNet block is been fully  pruned, what will happen?

**Comment by vtddggg** [comment]
sorry for the late reply, if search process is done, model definition, logs and checkpoint are saved at your log_dir. Then you can do test in the way stated in `Readme`
My result is 

05/22 09:45:24 PM Epoch 0/350 test: loss: 1.0999, top1_error: 0.2564, top5_error: 0.0825
2020-05-01 00:00:02
05/22 09:44:46 PM Epoch 0/350 bn_calibration: loss: 1.1393, top1_error: 0.2544, top5_error: 0.0950

I think it is matched with the result in papers.

**Comment by meijieru** [comment]
> @vtddggg
> Hi, Author! I'm sorry to bother you again. If a mobileNet block is been fully pruned, what will happen?

If the block has skip connection, it is OK.
If not, the training acc will go zero, thus this will not happen.

**Comment by vtddggg** [comment]
> > @vtddggg
> > Hi, Author! I'm sorry to bother you again. If a mobileNet block is been fully pruned, what will happen?
> 
> If the block has skip connection, it is OK.
> If not, the training acc will go zero, thus this will not happen.

So how to fix this AssertionError?

**Comment by meijieru** [comment]
> > > @vtddggg
> > > Hi, Author! I'm sorry to bother you again. If a mobileNet block is been fully pruned, what will happen?
> > 
> > 
> > If the block has skip connection, it is OK.
> > If not, the training acc will go zero, thus this will not happen.
> 
> So how to fix this AssertionError?

I will check it later.

**Comment by iFLYTEK-Embodied-Robotics-Team** [comment]
@meijieru @vtddggg 
Sorror, I don't occur the AssertionErro again, maybe i make a special mistake! And I have get the similar search results as your paper. Please close this comment~Thank you for your reply!
